### ISSUE TITLE
Load Pipeline sparkling from s3 in emr 

### ISSUE THREAD

**Issue by joscani** [h2oai/sparkling-water#2344]
Error loading s3 pipeline sparkling model previously saved
Info 

- Sparkling water sparkling-water-package_2.11-3.30.1.3-1-2.4.jar
- Execution mode `YARN-client`

I launch sparkling water using this code

```bash
spark-shell \
--jars /home/jcanadar/sparkling-water-package_2.11-3.30.1.3-1-2.4.jar \
--conf spark.executor.instances=2 \
--conf spark.executor.cores=4 \
--conf spark.executor.memory=8G \
--conf spark.driver.memory=2G \
--conf spark.kryo.registrationRequired=true \
--conf spark.scheduler.maxRegisteredResourcesWaitingTime=1000000 \
--conf spark.ext.h2o.fail.on.unsupported.spark.param=false \
--conf spark.dynamicAllocation.enabled=false \
--conf yarn.nodemanager.vmem-check-enabled=false \
--conf spark.sql.autoBroadcastJoinThreshold=-1 \
--conf spark.executor.memoryOverhead=1G \
--conf spark.locality.wait=30000 \
--conf spark.scheduler.minRegisteredResourcesRatio=1 \
--deploy-mode client \
--name check-targetencoding
```
I try to save and load a sparkling water  pipeline

```scala
import org.apache.spark.sql.{DataFrame, SaveMode}
import spark.implicits._
import org.apache.spark.sql.functions._
import org.apache.spark.ml.{Pipeline, PipelineModel}

import org.apache.spark.sql.functions._
import org.apache.spark.h2o._
import java.net.URI
val hc = H2OContext.getOrCreate()
import ai.h2o.sparkling.ml.features.H2OTargetEncoder
import ai.h2o.sparkling.ml.algos.H2OGBM
  

val training = spark.createDataFrame(Seq(
  (0L, "a b c d e spark", 1.0),
  (1L, "b d", 0.0),
  (2L, "spark f g h", 1.0),
  (2L, "spark f g h", 0.0),
  (2L, "spark f g h", 1.0),
  (3L, "hadoop mapreduce", 0.0)
)).toDF("id", "text", "label")


val targetEncoder = new H2OTargetEncoder().
  setInputCols(Array("text","id")).
  setLabelCol("label")

val pipeline = new Pipeline().setStages(Array(targetEncoder))
val pipelineModel = pipeline.fit(training)

pipelineModel.write.overwrite().save("s3://mybucket/target_encodingpipeline")

```
The pipeline is saved correctly but when I try to load from s3 fails

```scala
val pipelineh2o = PipelineModel.load("s3://mybucket/target_encodingpipeline")
```

And get the error

```bash
main  WARN org.apache.hadoop.conf.Configuration: core-site.xml:an attempt to override final parameter: fs.s3.buffer.dir;  Ignoring.
java.lang.NullPointerException: BucketName is required
  at com.amazon.ws.emr.hadoop.fs.shaded.com.google.common.base.Preconditions.checkNotNull(Preconditions.java:226)
  at com.amazon.ws.emr.hadoop.fs.s3.lite.executor.GlobalS3Executor.execute(GlobalS3Executor.java:85)
  at com.amazon.ws.emr.hadoop.fs.s3.lite.AmazonS3LiteClient.invoke(AmazonS3LiteClient.java:184)
  at com.amazon.ws.emr.hadoop.fs.s3.lite.AmazonS3LiteClient.doesBucketExist(AmazonS3LiteClient.java:90)
  at com.amazon.ws.emr.hadoop.fs.s3n.Jets3tNativeFileSystemStore.ensureBucketExists(Jets3tNativeFileSystemStore.java:139)
  at com.amazon.ws.emr.hadoop.fs.s3n.Jets3tNativeFileSystemStore.initialize(Jets3tNativeFileSystemStore.java:116)
  at com.amazon.ws.emr.hadoop.fs.s3n.S3NativeFileSystem.initialize(S3NativeFileSy
```

If I try the same with a simple spark mlpipeline it works. Anyway if I copy h2o pipeline from s3 to edge node and after to a hdfs path it works. I thing is something with spark conf

**Comment by mn-mikke** [comment]
@joscani That's an interesting problem I will try to replicate and will get back to you

**Comment by joscani** [comment]
Other warning in sparkling water at the beginning is 

```shell
WARN Configuration: core-site.xml:an attempt to override final parameter: fs.s3.buffer.dir;  Ignoring.
```

**Comment by mn-mikke** [comment]
@joscani This problem will fixed in `3.32.0.3-1` by https://github.com/h2oai/sparkling-water/pull/2407. Thanks a lot for reporting it!
